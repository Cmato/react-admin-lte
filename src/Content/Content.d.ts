import * as React from 'react';

export interface ContentProps {
  breadcrumb?: React.ReactNode;
  component?: React.ReactType<ContentProps>;
  headerComponent?: React.ReactType<ContentProps>;
  subtitle?: React.ReactNode;
  title?: React.ReactNode;
}

declare const Content: React.ComponentType<ContentProps>;

export default Content;
