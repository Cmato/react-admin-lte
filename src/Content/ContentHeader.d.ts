import * as React from 'react';

export interface ContentHeaderProps {
  component?: React.ReactType<ContentHeaderProps>;
  subtitle?: React.ReactNode;
  title?: React.ReactNode;
}

declare const ContentHeader: React.ComponentType<ContentHeaderProps>;

export default ContentHeader;
