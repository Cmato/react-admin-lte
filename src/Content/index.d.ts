export { default } from './Content';
export * from './Content';
export { default as ContentHeader } from './ContentHeader';
export * from './ContentHeader';
export { default as ContentWrapper } from './ContentWrapper';
export * from './ContentWrapper';
