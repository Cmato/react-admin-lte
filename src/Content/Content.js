import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Content = ({
  breadcrumb,
  children,
  className: classNameProp,
  component: Component,
  headerComponent: HeaderComponent,
  subtitle,
  title,
  ...other
}) => {
  const className = classNames({
    'content-wrapper': true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      <HeaderComponent className="content-header">
        <h1>
          {title}
          <small>{subtitle}</small>
        </h1>
        {breadcrumb}
      </HeaderComponent>
      <Component className="content">
        {children}
      </Component>
    </div>
  );
};

Content.propTypes = {
  /**
   * Breadcrumb to be placed on the right side of the content header.
   */
  breadcrumb: PropTypes.node,
  /**
   * The content of the content.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  headerComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /**
   * Subtitle of the content.
   */
  subtitle: PropTypes.node,
  /**
   * Title of the content.
   */
  title: PropTypes.node
};

Content.defaultProps = {
  component: 'section',
  headerComponent: 'section'
};

export default Content;
