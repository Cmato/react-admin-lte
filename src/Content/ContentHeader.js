import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ContentHeader = ({
  children,
  className: classNameProp,
  component: Component,
  subtitle,
  title,
  ...other
}) => {
  const className = classNames({
    'content-header': true
  }, classNameProp);

  return (
    <Component className={className} {...other}>
      <h1>
        {title}
        <small>{subtitle}</small>
      </h1>
      {children}
    </Component>
  );
};

ContentHeader.propTypes = {
  /**
   * Used to render a breadcrumb.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /**
   * Subtitle of content.
   */
  subtitle: PropTypes.node,
  /**
   * Title of content.
   */
  title: PropTypes.node
};

ContentHeader.defaultProps = {
  component: 'section'
};

export default ContentHeader;
