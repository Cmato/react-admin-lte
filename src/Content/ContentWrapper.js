import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ContentWrapper = ({
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'content-wrapper': true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

ContentWrapper.propTypes = {
  /**
   * The content of the content wrapper.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

export default ContentWrapper;
