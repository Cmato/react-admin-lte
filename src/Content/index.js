export { default } from './Content';
export { default as ContentHeader } from './ContentHeader';
export { default as ContentWrapper } from './ContentWrapper';
