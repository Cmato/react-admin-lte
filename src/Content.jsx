import React from 'react';
import classNames from 'classnames';

import ContentHeader from './ContentHeader';
import ContentBody from './ContentBody';

const Content = ({
  className,
  children,
}) => {
  const classes = {
    'content-wrapper': true,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

Content.Header = ContentHeader;
Content.Body = ContentBody;

export default Content;
