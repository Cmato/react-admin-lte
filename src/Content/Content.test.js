import React from 'react';
import { shallow } from 'enzyme';
import Content from './Content';

it('renders default', () => {
  expect(shallow(<Content><div /></Content>))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<Content component="div"><div /></Content>))
    .toMatchSnapshot();
});

it('renders title', () => {
  expect(shallow(<Content title="Super important"><div /></Content>))
    .toMatchSnapshot();
});

it('renders subtitle', () => {
  expect(shallow(<Content subtitle="But only a little"><div /></Content>))
    .toMatchSnapshot();
});

it('renders breadcrumb', () => {
  expect(shallow(<Content breadcrumb="nom nom nom"><div /></Content>))
    .toMatchSnapshot();
});

it('renders custom header component', () => {
  expect(shallow(<Content headerComponent="div"><div /></Content>))
    .toMatchSnapshot();
});
