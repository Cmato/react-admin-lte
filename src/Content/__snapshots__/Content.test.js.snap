// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders breadcrumb 1`] = `
<div
  className="content-wrapper"
>
  <section
    className="content-header"
  >
    <h1>
      <small />
    </h1>
    nom nom nom
  </section>
  <section
    className="content"
  >
    <div />
  </section>
</div>
`;

exports[`renders custom component 1`] = `
<div
  className="content-wrapper"
>
  <section
    className="content-header"
  >
    <h1>
      <small />
    </h1>
  </section>
  <div
    className="content"
  >
    <div />
  </div>
</div>
`;

exports[`renders custom header component 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="content-header"
  >
    <h1>
      <small />
    </h1>
  </div>
  <section
    className="content"
  >
    <div />
  </section>
</div>
`;

exports[`renders default 1`] = `
<div
  className="content-wrapper"
>
  <section
    className="content-header"
  >
    <h1>
      <small />
    </h1>
  </section>
  <section
    className="content"
  >
    <div />
  </section>
</div>
`;

exports[`renders subtitle 1`] = `
<div
  className="content-wrapper"
>
  <section
    className="content-header"
  >
    <h1>
      <small>
        But only a little
      </small>
    </h1>
  </section>
  <section
    className="content"
  >
    <div />
  </section>
</div>
`;

exports[`renders title 1`] = `
<div
  className="content-wrapper"
>
  <section
    className="content-header"
  >
    <h1>
      Super important
      <small />
    </h1>
  </section>
  <section
    className="content"
  >
    <div />
  </section>
</div>
`;
